app/controllers/admin/speeches_controller.rb

```diff
@@ -1,6 +1,4 @@
 class Admin::SpeechesController < Admin::EditionsController
-  before_action :clear_role_appointment_param_on_override, only: %i[update create]
```

Comment:
If you want to avoid putting speech-specific stuff in the main Editions controller, you could always extend the `clean_edition_parameters` method in Speeches controller instead.

```ruby
class Admin::SpeechesController < Admin::EditionsController
  def clean_edition_parameters
    super
    clear_role_appointment_param_on_override
  end

private

  # ...
end
```

I'm undecided whether it's better for this to live in Editions controller or in Speeches controller. I don't have a strong opinion either way – so feel free to ignore & leave it as is, if you think that's better.

Reply:
yea i think this is better 👍 Have updated it